src/drivers/serac.cpp

```diff
@@ -80,33 +80,44 @@ int main(int argc, char* argv[])
     input_file_path = search->second;
   }
 
+  // Check if a restart was requested
+  std::optional<int> restart_cycle;
+  if (auto cycle = cli_opts.find("restart_cycle"); cycle != cli_opts.end()) {
+    restart_cycle = std::stoi(cycle->second);
+  }
+
   // Check for the doc creation command line argument
   bool create_input_file_docs = cli_opts.find("create_input_file_docs") != cli_opts.end();
 
   // Create DataStore
   axom::sidre::DataStore datastore;
 
+  // Intialize MFEMSidreDataCollection
+  serac::StateManager::initialize(datastore, restart_cycle);
+
   // Initialize Inlet and read input file
   auto inlet = serac::input::initialize(datastore, input_file_path);
   serac::defineInputFileSchema(inlet, rank);
 
   // Optionally, create input file documentation and quit
   if (create_input_file_docs) {
-    auto writer = std::make_unique<axom::inlet::SphinxDocWriter>("serac_input.rst", inlet.sidreGroup());
-    inlet.registerDocWriter(std::move(writer));
+    auto writer = std::make_unique<axom::inlet::SphinxWriter>("serac_input.rst");
+    inlet.registerWriter(std::move(writer));
     inlet.writeDoc();
     serac::exitGracefully();
   }
 
   // Save input values to file
   datastore.getRoot()->save("serac_input.json", "json");
 
-  std::shared_ptr<mfem::ParMesh> mesh;
-  // Build the mesh
-  auto mesh_options = inlet["main_mesh"].get<serac::mesh::InputOptions>();
-  if (const auto file_opts = std::get_if<serac::mesh::FileInputOptions>(&mesh_options.extra_options)) {
-    auto full_mesh_path = serac::input::findMeshFilePath(file_opts->relative_mesh_file_name, input_file_path);
-    mesh = serac::buildMeshFromFile(full_mesh_path, mesh_options.ser_ref_levels, mesh_options.par_ref_levels);
+  if (!restart_cycle) {
```

Comment:
Add a comment here something like:

// Not restarting, mesh must be created and added to the StateManager